cirq-web/cirq_ts/src/bloch_sphere/main.ts

```diff
@@ -0,0 +1,45 @@
+// Copyright 2021 The Cirq Developers
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {BlochSphereScene} from './components/scene';
+import {BlochSphere} from './bloch_sphere';
+
+/**
+ * Adds a Bloch sphere element with relevant, configurable data for the
+ * sphere shape of the Bloch sphere and the state vector displayed with it.
+ * These elements are added to a Scene object, which is added to the DOM
+ * tree in the BlochSphereScene class.
+ * @param circleData A JSON string containing information that configures the
+ * bloch sphere.
+ * @param vectorData A JSON string containing information that configures the
+ * state vector.
+ * @param containerId A string containing the container (div, span, etc.) id that will contain the visualization
+ * output.
+ */
+export function blochSphere(
```

Comment:
I find the signature of this function confusing: 
- `circleData` - there is no circle - a better could be `blochSphereConfigJSON`? 
- `vectorData` - what vector? Typically you want to name things closer to their domain name - this case it is a _state vector_. 
- `data` is superflous, maybe you can use `JSON` to signify the format

The name of the method is `blochSphere` but it does two things: it creates a bloch sphere and then adds a vector to it. Consider one of the following options:
- rename to `renderBlochSphereWithStateVector()`
- rename to`renderBlochSphere()` and return the blochsphere instance on which then the client can call `blochSphere.addVector(...)`

Reply:
I like the second, specifically because it will make it easier for modify the code in the future to support adding multiple vectors cleanly (it allows for it right now, but the vectors are all the same color).

Reply:
Except wouldn't it need to be `addVector(blochSphere, stateVectorJSON)`? If we wanted create another BlochSphere class in main, we could go ahead and do that too.